src/jit/gcinfo.cpp

```diff
@@ -524,56 +463,128 @@ void GCInfo::gcCountForHeader(UNALIGNED unsigned int* untrackedCount, UNALIGNED
         {
-            /* Special case: skip any 0-length lifetimes */
+            // Special case: skip any 0-length lifetimes.
 
             if (varTmp->vpdBegOfs == varTmp->vpdEndOfs)
             {
                 continue;
             }
 
-            count++;
+            varPtrTableSize++;
         }
     }
 
 #ifdef DEBUG
     if (compiler->verbose)
     {
-        printf("GCINFO: trackdLcls = %u\n", count);
+        printf("GCINFO: trackdLcls = %u\n", varPtrTableSize);
     }
 #endif
 
-    *varPtrTableSize = count;
+    *pVarPtrTableSize = varPtrTableSize;
+}
+
+//------------------------------------------------------------------------
+// gcIsUntrackedLocalOrNonEnregisteredArg: Check if this varNum with GC type
+// corresponds to an untracked local or argument that was not fully enregistered.
+//
+//
+// Arguments:
+//   varNum - the variable number to check;
+//   pThisKeptAliveIsInUntracked - if !WIN64EXCEPTIONS and the argument != nullptr remember if `this` should be
+//   untracked and kept alive.
+//
+// Return value:
+//   true if it an untracked pointer value.
+//
+bool GCInfo::gcIsUntrackedLocalOrNonEnregisteredArg(unsigned varNum, bool* pThisKeptAliveIsInUntracked)
+{
+    LclVarDsc* varDsc = compiler->lvaGetDesc(varNum);
+
+    assert(!compiler->lvaIsFieldOfDependentlyPromotedStruct(varDsc));
+    assert(varTypeIsGC(varDsc->TypeGet()));
+
+    // Do we have an argument or local variable?
+    if (!varDsc->lvIsParam)
+    {
+        // If is pinned, it must be an untracked local.
+        assert(!varDsc->lvPinned || !varDsc->lvTracked);
+
+        if (varDsc->lvTracked || !varDsc->lvOnFrame)
+        {
+            return false;
+        }
+    }
+    else
+    {
+        // Stack-passed arguments which are not enregistered are always reported in this "untracked stack pointers"
+        // section of the GC info even if lvTracked==true.
+
+        // Has this argument been fully enregistered?
+        if (!varDsc->lvOnFrame)
+        {
+            // If a CEE_JMP has been used, then we need to report all the arguments even if they are enregistered, since
+            // we will be using this value in JMP call.  Note that this is subtle as we require that argument offsets
+            // are always fixed up properly even if lvRegister is set .
+            if (!compiler->compJmpOpUsed)
+            {
+                return false;
+            }
+        }
+        else if (varDsc->lvIsRegArg && varDsc->lvTracked)
+        {
+            // If this register-passed arg is tracked, then it has been allocated space near the other pointer variables
+            // and we have accurate life-time info. It will be reported with gcVarPtrList in the "tracked-pointer"
+            // section.
+            return false;
+        }
+    }
+
+#if !defined(FEATURE_EH_FUNCLETS)
+    // For FEATURE_EH_FUNCLETS, "this" must always be in untracked variables
+    // so we cannot have "this" in variable lifetimes
```

Comment:
Again, this is confusing - it says that "this" must always be in untracked variables, but below it says that it is being reported as a tracked variable. I *think* this means that it will always be in the untracked variable area, but will have to be reported as tracked if we have `lvaKeepAliveAndReportThis()`. Is that the case?